File: frontend/lib/model/subcategory.dart
import 'dart:convert';

// ignore_for_file: public_member_api_docs, sort_constructors_first
class SubCategory {
  String categoryId;
  String categoryName;
  String image;
  String subcategoryName;
  SubCategory({
    required this.categoryId,
    required this.categoryName,
    required this.image,
    required this.subcategoryName,
  });

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'categoryId': categoryId,
      'categoryName': categoryName,
      'image': image,
      'subcategoryName': subcategoryName,
    };
  }

  factory SubCategory.fromJson(Map<String, dynamic> map) {
    return SubCategory(
      categoryId: map['categoryId'] as String,
      categoryName: map['categoryName'] as String,
      image: map['image'] as String,
      subcategoryName: map['subcategoryName'] as String,
    );
  }

  String toJson() => json.encode(toMap());

 
}

File: backend/models/subcategory.js
const mongoose = require('mongoose');

const subcategorySchema = new mongoose.Schema({
    categoryId:{
        type:String,
        required:true
    },
    categoryName:{
        type:String,
        required:true
    },
    image:{
        type:String,
        required:true
    },
    subcategoryName:{
        type:String,
        required:true
    }
});

const Subcategory = mongoose.model('Subcategory',subcategorySchema);
module.exports = Subcategory;